### PR TITLE
Notify client when a move is a spread move

### PR DESCRIPTION
**Pull request by QuiteQuiet**
Based of what was said in https://github.com/Zarel/Pokemon-Showdown-Client/pull/662#discussion_r54695369 you were looking for something like this.

### CODE REVIEW

File: data/scripts.js [hunk 1]
```diff
@@ -140,7 +140,10 @@ exports.BattleScripts = {
 				this.add('-notarget');
 				return true;
 			}
-			if (targets.length > 1) move.spreadHit = true;
+			if (targets.length > 1) {
+				this.attrLastMove('[spread]');
+				move.spreadHit = true;
+			}
```
Comment (Zarel) [inline data/scripts.js]: Even better, though, would be to move this below, and send the list of successful targets in `[spread]`

Comment (QuiteQuiet) [inline data/scripts.js]: So something like this instead?
````
let hitTargets = [];
for (let i = 0; i < targets.length; i++) {
    let hitResult = this.tryMoveHit(targets[i], pokemon, move, true);
    if (hitResult || hitResult === 0 || hitResult === undefined) {
        moveResult = true;
        hitTargets.push(targets[i]);
    }
    damage += hitResult || 0;
}
if (move.spreadHit) this.attrLastMove('[spread] ' + hitTargets.join(','));
````
(character separating each target probably shouldn't be `,` but the general idea?)

Comment (Zarel) [inline data/scripts.js]: Yes.

Comment (Zarel) [inline data/scripts.js]: I would actually just use the target's location ID (`p1a`, `p1b`, etc) so we could use `,` as the separator.

Comment (QuiteQuiet) [inline data/scripts.js]: The only place I can find where `p1a`/`p1b`/others are made is in `BattlePokemon#toString` and it doesn't seem to be stored anywhere else. Does that mean `targets[i].toString().substr(0, 3)` (or something along those lines) is the only way to get the location ID?

Comment (Zarel) [inline data/scripts.js]: Not the only way, but it might be the easiest way.

**Pull request merged**